src/createServer.js

```diff
@@ -1,11 +1,186 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
   // Use express to create a server
   // Add a routes to the server
   // Return the server (express app)
+  const app = express();
+
+  let usersData = [];
+  let exprensesData = [];
+
+  app.delete('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+    const filteredData = usersData.filter(user => user.id !== +userId);
+
+    if (filteredData.length === usersData.length) {
+      res.sendStatus(404);
+    } else {
+      usersData = filteredData;
+      res.status(204);
+      res.send(usersData);
+    }
+  });
+
+  app.patch('/users/:userId', express.json(), (req, res) => {
+    const { userId } = req.params;
+    const foundUserId = usersData.find(user => user.id === +userId);
+
+    if (!foundUserId) {
+      res.sendStatus(404);
+    } else {
+      const { name } = req.body;
+
+      Object.assign(foundUserId, { name });
+
+      res.send(foundUserId);
+    }
+  });
+
+  app.get('/users', (req, res) => {
+    if (!usersData) {
+      res.send([]);
+    } else {
+      res.setHeader('Content-Type', 'application/json');
+      res.status(200);
+      res.send(usersData);
+    }
+  });
+
+  app.get('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+    const foundUserId = usersData.find(user => user.id === +userId);
+
+    if (!foundUserId) {
+      res.status(404);
+    }
+
+    res.send(foundUserId);
+  });
+
+  app.post('/users', express.json(), (req, res) => {
+    const { name } = req.body;
+
+    const newUser = {
+      id: Math.random(),
+      name,
+    };
+
+    if (!name) {
+      res.status(400);
+      res.end();
+    } else {
+      usersData.push(newUser);
+      res.setHeader('Content-Type', 'application/json');
+      res.status(201);
+      res.send(newUser);
+    }
+  });
+
+  app.post('/expenses', express.json(), (req, res) => {
+    const { userId } = req.body;
+
+    const foundUserId = usersData.find(user => user.id === +userId);
+
+    if (!foundUserId) {
+      res.status(400);
+      res.end();
+    } else {
+      const newExprenses = {
+        id: Math.random(),
+        ...req.body,
+      };
+
+      exprensesData.push(newExprenses);
+      res.setHeader('Content-Type', 'application/json');
+      res.status(201);
+      res.send(newExprenses);
+    }
+  });
+
+  app.get('/expenses', (req, res) => {
```

Comment:
generally try to restructure your code so that there are fewer complex conditions. It's hard to read now